File: libs/eave-stdlib-ts/src/core-api/models/integrations.ts
import { AtlassianInstallationPeek } from "./atlassian.js";
import { ConnectInstallationPeek } from "./connect.js";
import { GithubInstallationPeek } from "./github.js";
import { SlackInstallationPeek } from "./slack.js";

/**
 * Key-value mapping of Integration to Installation info.
 * The keys here will match the enum cases in enums.Integration
 */
export type Integrations = {
  github_integration: GithubInstallationPeek | null;
  slack_integration: SlackInstallationPeek | null;
  atlassian_integration: AtlassianInstallationPeek | null;
  confluence_integration: ConnectInstallationPeek | null;
  jira_integration: ConnectInstallationPeek | null;
};

File: libs/eave-stdlib-py/src/eave/stdlib/core_api/models/integrations.py
import enum
from typing import Optional

from eave.stdlib.core_api.models.team import ConfluenceDestination
from .atlassian import AtlassianInstallation, AtlassianInstallationPeek
from .github import GithubInstallation, GithubInstallationPeek
from .slack import SlackInstallation, SlackInstallationPeek
from .connect import ConnectInstallation, ConnectInstallationPeek
from . import BaseResponseModel


class Integration(enum.StrEnum):
    """
    Apps that a Team can integrate with.
    """

    slack = "slack"
    github = "github"
    atlassian = "atlassian"
    confluence = "confluence"
    jira = "jira"


class Integrations(BaseResponseModel):
    github_integration: Optional[GithubInstallationPeek]
    slack_integration: Optional[SlackInstallationPeek]
    atlassian_integration: Optional[AtlassianInstallationPeek]
    confluence_integration: Optional[ConnectInstallationPeek]
    jira_integration: Optional[ConnectInstallationPeek]
